src/black/trans.py

```diff
@@ -1243,6 +1243,45 @@ def more_splits_should_be_made() -> bool:
             last_line.comments = line.comments.copy()
             yield Ok(last_line)
 
+    def _get_nameescape_slices(self, string: str) -> List[Tuple[int, int]]:
+        """
+        Returns:
+            List of all ranges of @string which, if @string were to be split there,
+            would result in the splitting of an \\N{...} expression (which is NOT
+            allowed).
+        """
+        slices = []
+        # None - looking for first backslash
+        # True - the previous backslash was unescaped
+        # False - the previous backslash was escaped
+        backslash_tracking_state: Optional[bool] = None
+        it = iter(enumerate(string))
+        for idx, c in it:
+            if backslash_tracking_state is None:
+                if c == "\\":
+                    backslash_tracking_state = True
+                continue
```

Comment:
Is this redundant? Because `not None is True`, so this could be combined with below. I think. Then it would start flipping between `True` and `False`.

Reply:
It isn't because we want to do an early-continue *if* it's not a backslash.

Reply:
I get that you're signifying "no backslashes" with `None`, but `False` seems to do the same thing, since we don't care where the escaping happens. But at this point it's splitting hairs 😅 so feel free to ignore me.

Reply:
Although I guess I could change the order and simplify this a little bit:
```py
if c == "\\":
    backslash_tracking_state = not backslash_tracking_state
    continue
if backslash_tracking_state is None:
    continue
```

Reply:
Okay, I see what you mean now 😄 Pushed a commit that should simplify this a fair bit.